packages/shopify-app-remix/src/server/authenticate/admin/authenticate.ts

```diff
@@ -104,124 +111,51 @@ export class AuthStrategy<
     }
   }
 
+  private async handleRoutes(request: Request) {
+    await this.handleBouncePageRoute(request);
+    await this.handleExitIframeRoute(request);
+    await this.strategy.handleRoutes(request);
+  }
+
   private async authenticateAndGetSessionContext(
     request: Request,
   ): Promise<SessionContext> {
-    const {api, logger, config} = this;
-    const params: BasicParams = {api, logger, config};
-
-    const url = new URL(request.url);
-
-    const isPatchSessionToken =
-      url.pathname === config.auth.patchSessionTokenPath;
-    const isExitIframe = url.pathname === config.auth.exitIframePath;
-    const isAuthRequest = url.pathname === config.auth.path;
-    const isAuthCallbackRequest = url.pathname === config.auth.callbackPath;
-    const sessionTokenHeader = getSessionTokenHeader(request);
-
+    const {logger} = this;
     logger.info('Authenticating admin request');
 
-    if (isPatchSessionToken) {
-      logger.debug('Rendering bounce page');
-      throw renderAppBridge(params, request);
-    } else if (isExitIframe) {
-      const destination = url.searchParams.get('exitIframe')!;
-
-      logger.debug('Rendering exit iframe page', {destination});
-      throw renderAppBridge(params, request, {url: destination});
-    } else if (isAuthCallbackRequest) {
-      throw await this.handleAuthCallbackRequest(request);
-    } else if (isAuthRequest) {
-      throw await this.handleAuthBeginRequest(request);
-    } else if (sessionTokenHeader) {
-      const sessionToken = await validateSessionToken(
-        {api, logger, config},
-        sessionTokenHeader,
-      );
+    const sessionTokenHeader = getSessionTokenHeader(request);
 
-      return this.validateAuthenticatedSession(request, sessionToken);
-    } else {
+    if (!sessionTokenHeader) {
       await this.validateUrlParams(request);
```

Comment:
Both the session token and the URL params expire and can only be generated using the API secret.  Given that, does token exchange need to validateURL Params?